include/pythontarget.h

```diff
@@ -184,6 +184,53 @@ typedef struct {
 } generic_port_capsule_struct;
 
 
+/**
+ * Special version of the template_input_output_port_struct
+ * for dynamic arrays.
+ * FIXME: This is currently kept for compatibility reasons
+ * and should be removed soon.
+ **/
+typedef struct {
+    PyObject_HEAD
+    PyObject* value;
+    bool is_present;
+    int num_destinations;
+    lf_token_t* token;
+    int length;
+} generic_port_instance_with_token_struct;
+
+/**
+ * Python wrapper for the tag_t struct in the C target.
+ **/
+typedef struct {
+    PyObject_HEAD
+    tag_t tag;
+} py_tag_t;
+
+PyTypeObject TagType;
+
+/**
+ * Tag getter for the "time" attribute
+ **/
+static PyObject* Tag_get_time(py_tag_t *self, void *closure);
+
+/**
+ * Tag getter for the "microstep" attribute
+ **/
+static PyObject* Tag_get_microstep(py_tag_t *self, void *closure);
+
+/**
+ * Initialize the Tag object with the given values for "time" and "microstep", 
+ * both of which are required.
+ * same as __init__()
+ * 
+ * @param self A py_tag_t object.
+ * @param args The arguments are:
+ *      @param time A logical time.
```

Comment:
Weird indentation artifact here.

Reply:
the indentation is used to show that "args" has 2 arguments, and what each argument represents

Reply:
Ah okay. Probably `@param` might not be the right tag here. I think there is a risk that time and microstep show up as function parameters in the doc. Maybe just use `-` at the beginning of each line?

Reply:
sounds good.